Group-09/EventManager/EventManager.hpp

```
@@ -1,48 +1,194 @@
+   */
+  void StartQueue() noexcept{
+    if (!running_) {
+      running_ = true;
+      start_time_ = std::chrono::steady_clock::now();
+    }
+  }
+
+  /**
+   * @brief Restarts clock
+   */
+  void RestartQueue() {
+    StopQueue();
+    total_runtime_ = std::chrono::duration<double>::zero();
+    StartQueue();
+  }
+
+  /**
+   * @brief Adds event to event_queue
+   * @param time The time in seconds when the event should be triggered
+   * @param func The function to be called when the event is triggered
+   * @param args The arguments to be passed to the function
+   * @return added Event
+   */
+  std::optional<Event<Args...>> AddEvent(int time,
+                                         std::function<void(Args...)> func,
+                                         Args... args) {
+    if (time < 0) {
+      throw std::invalid_argument("Time must be positive");
+    }
+    Event<Args...> event(next_id_, time, func, std::forward<Args>(args)...);
+    ++next_id_;
+    event_queue_.add(event);
+    running_events_.insert(event.getID());
+    return event;
+  }
+
+  /**
+   * @brief Adds event to paused_events
+   * @param event The event to be paused
+   * @return true if successful, false if event does not exist in queue
+   */
+  bool PauseEvent(const Event<Args...> &event) {
+    int id = event.getID();
+    assert(((paused_events_.find(id) != paused_events_.end()) ||
+            (running_events_.count(id) > 0)) &&
+           "Event ID must be a managed ID.");
+    if (paused_events_.count(id) > 0) {
+      return true;
+    } else if (running_events_.count(id)) {
+      paused_events_.insert({id, event});
+      running_events_.erase(id);
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * @brief Removes event to paused_events set
+   * @param event The event to be resumed
+   * @return true if successfully removed, false if event is not currently
+   * paused
+   */
+  bool ResumeEvent(Event<Args...> &event) {
```

Similar to PauseEvent, this is only guarded with an assert. Resume on an invalid ID is silently ignored in release mode.